### PR TITLE
Add ti::real C++ class wrapping TI's real number format and OS funcs

### PR DESCRIPTION
**Pull request by adriweb**
**TODO**:

 * [x] Better math-oriented example?
 * [x] Autotest on that example
 * [x] wrap the remaining functions

### CODE REVIEW

**Comment by adriweb** [comment]
We could (should) probably come up with a fancy math example though.

**Comment by erentar** [comment]
Please put the monte carlo pi as math example

**Comment by adriweb** [comment]
Sure, that's easy enough

**Comment by adriweb** [comment]
OK, review plz, I added more functions and tests.

File: src/ce/tireal.hpp [hunk 1]
```diff
@@ -0,0 +1,585 @@
+#pragma once
```
Comment (mateoconlechuga) [inline src/ce/tireal.hpp:1]: no

File: src/ce/tireal.hpp [hunk 2]
```diff
@@ -0,0 +1,585 @@
...
+
+#include <tice.h>
+
+#include "limits.hpp"
```
Comment (mateoconlechuga) [inline src/ce/tireal.hpp:5]: should be the very first include

File: src/std/limits.hpp [hunk 3]
```diff
@@ -0,0 +1,29 @@
+#pragma once
```
Comment (mateoconlechuga) [inline src/std/limits.hpp:1]: no

**Pull request merged**